### ISSUE TITLE
Add support for multiple predicates

### ISSUE THREAD

**Issue by talios** [sviperll/adt4j#17]
I have a case where ideally I want to have multiple predicates defined for a value.

I have a `ResultVisitor` with states such as `succeed`, `succeedWithValue`, `noop`, `failure`.  And want to have predicates like `isSuccess` (noop, suceed, succeedWithValue) and `hasRollback` (suceed, suceedWithValue ).

Currently I have to split these with an `isFailure` on the `failure` case.

Thoughts?

**Comment by talios** [comment]
An alternative to this I've also been thinking of, is adding support for `Optional` style `@Getter`s.

Currently, a getter will generate a method that with returns the value, or throws an exception for the invalid cases. In a bunch of cases I find I wrap these with predicates, but if the function returned a `Some` or `None` then one could write code like:

    value.rollback().ifPresent( r -> ...... );

If something like this was added then support for the various Optional style libraries would be good: Google Guava, JDK8, and javaslang would probably be the main ones that come to mind.

**Comment by sviperll** [comment]
Are you ok with multiple predicates like this:

```java
@GenerateValueClassForVisitor
@Visitor(resultVariableName="R")
interface ResultVisitor<R> {
    @GeneratePredicates(
        @GeneratePredicate(name="isSuccess"),
        @GeneratePredicate(name="hasRollback"))
    R succeed(...);

    @GeneratePredicates(
        @GeneratePredicate(name="isSuccess"),
        @GeneratePredicate(name="hasRollback"))
    R succeedWithValue(...);

    @GeneratePredicate(name="isSuccess")
    R noop(...);

    R failure(...);
}
```

**Comment by talios** [comment]
@sviperll That'd be fine. I know JDK8 allows for multiple standalone `@GeneratePredicate` now, but in the mean time - that's good.

**Comment by sviperll** [comment]
Just released 1.2 version with multiple predicates support.

**Comment by talios** [comment]
+1000 - you're a legend!
